scripts/template_upload_gce.py

```diff
@@ -67,33 +73,54 @@ def make_kwargs(args, **kwargs):
     return kwargs
 
 
-def download_image_file(image_url, destination=None):
+def make_ssh_client(ssh_host, ssh_user, ssh_pass):
+    connect_kwargs = {
+        'username': ssh_user,
+        'password': ssh_pass,
+        'hostname': ssh_host
+    }
+    return SSHClient(**connect_kwargs)
+
+
+def log_detail(message, provider=None):
+    if provider:
+        print("{} GCE: {}: {}".format(datetime.utcnow(), provider, message))
+    else:
+        print("{} INFO: {}".format(datetime.utcnow(), message))
+
+
+def download_image_file(image_url, ssh_client):
+    """
+    Download the file to the cli-tool-client and return the file path + file name
+    Default destinations are None and are set to the cli-tool-client if left that way
+    :param image_url: string url to the tar.gz image
+    :return: tuple: file_name, file_path strings
+    """
+    # TODO: add a --local option to the script and run commands locally to download
+    # Download to the amazon/gce upload machine in the lab because its fast
+    target_dir = '/root/downloads'
     file_name = image_url.split('/')[-1]
-    u = urllib2.urlopen(image_url)
-    meta = u.info()
-    file_size = int(meta.getheaders("Content-Length")[0])
-    file_path = os.path.abspath(file_name)
-    if os.path.isfile(file_name):
-        if file_size == os.path.getsize(file_name):
-            return (file_name, file_path)
-        os.remove(file_name)
-    print("Downloading: {} Bytes: {}".format(file_name, file_size))
-    with open(file_name, 'wb') as f:
-        file_size_dl = 0
-        block_sz = 8192
-        while True:
-            buffer_f = u.read(block_sz)
-            if not buffer_f:
-                break
-
-            file_size_dl += len(buffer_f)
-            f.write(buffer_f)
-            status = r"{:.2f}".format(file_size_dl, file_size_dl * 100. / file_size)
-            status += chr(8) * (len(status) + 1)
-            print(status)
-            sleep(5)
-
-    return (file_name, file_path)
+    # check if file exists
+    print('INFO: Checking if file exists on cli-tool-client...')
+    result = ssh_client.run_command('ls -1 {}/{}'.format(target_dir, file_name))
```

Comment:
Did this everywhere that I had concatenated {}/{}, for local paths and those on the GCE bucket.